beacon-chain/rpc/eth/beacon/validator.go

```diff
@@ -235,6 +242,117 @@ func (bs *Server) ListCommittees(ctx context.Context, req *ethpb.StateCommittees
 	return &ethpb.StateCommitteesResponse{Data: committees, ExecutionOptimistic: isOptimistic, Finalized: isFinalized}, nil
 }
 
+// GetValidatorCount is a HTTP handler that serves the GET /eth/v1/beacon/states/{state_id}/validator_count endpoint.
+// It returns the total validator count according to the given status provided as a query parameter.
+//
+// The state ID is expected to be a valid Beacon Chain state identifier.
+// The status query parameter should be one of the following values: pending_initialized, pending_queued, active_ongoing,
+// active_exiting, active_slashed, exited_unslashed, exited_slashed, withdrawal_possible, withdrawal_done, active, pending, exited, withdrawal.
+// The response is a JSON object containing the total validator count for the specified status.
+//
+// Example usage:
+//
+//	GET /eth/v1/beacon/states/12345/validator_count?status=active
+//
+// The above request will return a JSON response like:
+//
+//	{
+//	 "execution_optimistic": "false",
+//	 "finalized": "true",
+//	 "data": {
+//	  "validator_count": "1024"
+//	 }
+//	}
+func (bs *Server) GetValidatorCount(w http.ResponseWriter, r *http.Request) {
+	stateID := mux.Vars(r)["state_id"]
```

Comment:
could you add a span trace here for this endpoint?

Reply:
Done!